ext/DifferentiationInterfaceTapirExt/mutating.jl

```diff
@@ -0,0 +1,10 @@
+function DI.value_and_pullback!!(f!, y, dx, ::AutoTapir, x, dy, extras::Nothing)
+    rrule = build_rrule(f!, y, x)
+    dy_righttype = convert(typeof(y), dy)
```

Comment:
This should use `Tapir.tangent_type`, rather than converting it to be the same type.